### PR TITLE
Add a new API to parameterise the API endpoint for geolocation search

### PR DESCRIPTION
**Pull request by samgiles**
r? @wilsonpage

### CODE REVIEW

File: src/main/java/org/mozilla/magnet/scanner/MagnetScanner.java [hunk 1]
```diff
@@ -61,13 +61,21 @@ MagnetScanner useMdns(ScannerMdns scanner) {
         return this;
     }
 
+    public MagnetScanner useGeolocation() {
+        if (!mScanners.containsKey(ScannerGeolocation.class.getName())) {
+            mScanners.put(ScannerGeolocation.class.getName(), new ScannerGeolocation(mContext));
+        }
+
+        return this;
+    }
```
Comment (wilsonpage) [inline src/main/java/org/mozilla/magnet/scanner/MagnetScanner.java:70]: Would this work without the `apiUrl`?

Comment (samgiles) [reply src/main/java/org/mozilla/magnet/scanner/MagnetScanner.java:70]: Yes because the constructor is overloaded to support the current API too.

**Comment by wilsonpage** [comment]
OK so this patch makes the magnet-service beacon-search endpoint optionally configurable? Is this so that you can use a dev server for debugging?

r+

**Comment by samgiles** [comment]
> OK so this patch makes the magnet-service beacon-search endpoint optionally configurable? Is this so that you can use a dev server for debugging?

Yup!

**Pull request merged**